test/docfx.Test/lib/JsonUtilityTest.cs

```diff
@@ -554,6 +554,37 @@ public void TestJsonMerge(string a, string b, string result)
             Assert.Equal(result.Replace('\'', '\"'), container.ToString(Formatting.None));
         }
 
+        [Fact]
+        public void NullValueHasSourceInfo()
+        {
+            var (_, json) = JsonUtility.Parse("{'a': null,'b': null}".Replace('\'', '"'), "file");
+            var (_, obj) = JsonUtility.ToObject<ClassWithSourceInfo>(json);
+
+            Assert.NotNull(obj?.A);
+
+            Assert.Null(obj.A.Value);
+            Assert.Equal(1, obj.A.Line);
+            Assert.Equal(10, obj.A.Column);
+            Assert.Equal("file", obj.A.File);
+
+            Assert.Equal("b", obj.B.Value);
+            Assert.Equal(1, obj.B.Line);
+            Assert.Equal(20, obj.B.Column);
+            Assert.Equal("file", obj.B.File);
+        }
+
+        [Fact]
+        public void JsonWithTypeErrorsHasCorrectSourceInfo()
+        {
+            var (_, json) = JsonUtility.Parse(
+                "{'next': {'a': ['a1','a2'],'b': 'b'}, 'c': 'c'}".Replace('\'', '"'), "file");
+            var (errors, obj) = JsonUtility.ToObject<ClassWithSourceInfo>(json);
+
+            Assert.Null(obj.Next.A.Value);
```

Comment:
this test showcased that if we simply `return null` without skipping the token in a complex structure, the return object may corrupt. That is why _Report error for invalid yaml header value type as array_ failed when we simply not ignore nulls.

Reply:
No.
With "NullValueHandler = Ignore" This test will pass. I believe IgnoreNullValueHandler did the same thing as you did in this PR.

![image](https://user-images.githubusercontent.com/19990166/58528844-eed91a00-820a-11e9-89d0-4553e3b30368.png)

Reply:
@live1206